lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -509,6 +511,64 @@ func (c *ContainerProxy) StreamContainerLogs(name string, out io.Writer, started
 	return nil
 }
 
+// StreamContainerStats will provide a stream of container stats written to the provided
+// io.Writer.  Prior to writing to the provided io.Writer there will be a transformation
+// from the portLayer representation of stats to the docker format
+func (c *ContainerProxy) StreamContainerStats(ctx context.Context, id string, out io.Writer, stream bool, CPUMhz int64) error {
+	defer trace.End(trace.Begin(id))
+
+	plClient, transport := c.createNewAttachClientWithTimeouts(attachConnectTimeout, 0, attachAttemptTimeout)
+	defer transport.Close()
+
+	// create a child context that we control
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	params := containers.NewGetContainerStatsParamsWithContext(ctx)
+	params.ID = id
+	params.Stream = stream
+
+	// converter config
+	config := convert.ContainerStatsConfig{
+		Ctx:         ctx,
+		Cancel:      cancel,
+		VchMhz:      CPUMhz,
+		Stream:      stream,
+		ContainerID: id,
+		Out:         out,
+	}
+	// create our converter
+	containerConverter := convert.NewContainerStats(config)
+	// provide the writer for the portLayer and start listening for metrics
+	writer := containerConverter.Listen()
+	if writer == nil {
+		// problem with the listener
+		return InternalServerError(fmt.Sprintf("unable to gather container(%s) statistics", id))
+	}
+
+	_, err := plClient.Containers.GetContainerStats(params, writer)
+	if err != nil {
+		switch err := err.(type) {
+		case *containers.GetContainerStatsNotFound:
+			return NotFoundError(id)
+		case *containers.GetContainerStatsInternalServerError:
+			return InternalServerError("Server error from the interaction port layer")
+		default:
+			if ctx.Err() == context.Canceled {
+				return nil
+			}
+			//Check for EOF.  Since the connection, transport, and data handling are
+			//encapsulated inside of Swagger, we can only detect EOF by checking the
+			//error string
+			if strings.Contains(err.Error(), swaggerSubstringEOF) {
```

Comment:
Not sure -- again just following other patterns of interacting with a stream and swagger...